### PR TITLE
Support for building EKA2L1 on MinGW

### PR DESCRIPTION
**Pull request by Florastamine**
WIP, don't merge yet. This patch fixes a few issues preventing EKA2L1 to be compiled on Windows using a non-MSVC compiler (MinGW, for example).

### CODE REVIEW

**Comment by pent0** [comment]
Thank you very much! You pretty much rewrite external dependency lol.

Please follow the commit naming rules <category>: <description>.

For example, with first commit. external/filediag: fix compile errors on mingw or something like that

Second commit like common/virtualmem: dont forcibly....

And so on. Thank you very much again!

**Comment by Florastamine** [comment]
> You pretty much rewrite external dependency lol.
I just updated the dependency to [the latest master](https://github.com/mlabbe/nativefiledialog), which fixed the I reported in Discord. No need for hacky hacks :)

File: src/emu/console/src/main.cpp [hunk 1]
```diff
@@ -57,10 +57,17 @@
...
-#if EKA2L1_PLATFORM(WIN32)
-extern "C" _declspec(dllexport) DWORD NvOptimusEnablement = 0x00000001;
+#if defined(_MSC_VER)
+#define EKA2L1_EXPORT __declspec(dllexport)
+#else
+#define EKA2L1_EXPORT __attribute__((dllexport))
 #endif
 
+extern "C" {
+    EKA2L1_EXPORT DWORD NvOptimusEnablement = 0x00000001;
```
Comment (pent0) [inline src/emu/console/src/main.cpp]: These are only available on WIN32. Use macro EKA2L1_PLATFORM(WIN32) to explicitly declare them.

Comment (Florastamine) [reply src/emu/console/src/main.cpp]: Nice catch.
I'll try to wrap up the PR tonight if time permits.

**Comment by Florastamine** [comment]
Should be good now, although more testing always favorable. MinGW with GCC x86-64 9.2
image

**Comment by pent0** [comment]
Thanks, please also rename previous commits.

**Comment by pent0** [comment]
Thank you very much. :D

Notenotenote: Next time after rebase fetch upstream and use git rebase upstream/master. The 
```
Merge branch 'master' of https://github.com/Florastamine/EKA2L1
```
should never appear again :D

**Pull request merged**